src/WebJobs.Script/OutOfProc/Http/DefaultHttpWorkerService.cs

```diff
@@ -0,0 +1,178 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Net.Http.Formatting;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc.WebApiCompatShim;
+using Microsoft.AspNetCore.WebUtilities;
+using Microsoft.Azure.WebJobs.Script.Description;
+using Microsoft.Azure.WebJobs.Script.Extensions;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Newtonsoft.Json;
+
+namespace Microsoft.Azure.WebJobs.Script.OutOfProc.Http
+{
+    public class DefaultHttpWorkerService : IHttpWorkerService
+    {
+        private readonly HttpClient _httpClient;
+        private readonly HttpWorkerOptions _httpWorkerOptions;
+        private readonly ILogger _logger;
+
+        public DefaultHttpWorkerService(IOptions<HttpWorkerOptions> httpWorkerOptions, ILoggerFactory loggerFactory)
+        {
+            _httpClient = new HttpClient();
+            _httpWorkerOptions = httpWorkerOptions.Value;
+            _logger = loggerFactory.CreateLogger<DefaultHttpWorkerService>();
+        }
+
+        internal DefaultHttpWorkerService(HttpClient httpClient, IOptions<HttpWorkerOptions> httpWorkerOptions, ILoggerFactory loggerFactory)
+        {
+            _httpClient = httpClient;
+            _httpWorkerOptions = httpWorkerOptions.Value;
+            _logger = loggerFactory.CreateLogger<DefaultHttpWorkerService>();
+        }
+
+        public Task InvokeAsync(ScriptInvocationContext scriptInvocationContext)
+        {
+            if (scriptInvocationContext.FunctionMetadata.IsSimpleHttpTriggerFunction())
+            {
+                return ProcessHttpInAndOutInvocationRequest(scriptInvocationContext);
+            }
+            return ProcessDefaultInvocationRequest(scriptInvocationContext);
+        }
+
+        internal async Task ProcessHttpInAndOutInvocationRequest(ScriptInvocationContext scriptInvocationContext)
```

Comment:
Could this be renamed `ProcessSimpleHttpTriggerFunction`? I assume we'd all know then that "Simple" means an HttpTrigger with one input and one output? Maybe add a comment here to explain that :-)

Reply:
previous iteration had this as simple. Renamed to after feedback :)